bcipy/helpers/triggers.py

```diff
@@ -784,33 +793,13 @@ def load(path: str,
         """
 
         # Checking for file with given path, with or without .txt
-        triggers_list, system_offset = TriggerHandler.read_text_file(path)
-
-        try:
-            if exclusion:
-                for trigger_type in exclusion:
-                    triggers_list[:] = [item for item in triggers_list if not trigger_type.value == item[1]]
-
-            # apply system and provided offsets
-            for item in triggers_list:
-                item[2] = float(item[2]) + offset + system_offset
-        except Exception as e:
-            raise BciPyCoreException(
-                f'Invalid triggers.txt format error=[{e}] triggers=[{triggers_list}]')
-
-        new_trigger_list = []
-        for trigger in triggers_list:
-            try:
-                new_trigger_list.append(
-                    Trigger(trigger[0],
-                            TriggerType(trigger[1]),
-                            float(trigger[2])
-                            )
-                )
-            except Exception as e:
-                raise BciPyCoreException(f'{trigger} read from {path} is not able to be cast')
-
-        return new_trigger_list
+        triggers, system_offset = TriggerHandler.read_text_file(path)
+        excluded_types = exclusion or []
+        total_offset = offset + system_offset
```

Comment:
I like how succinct this is, however the custom exceptions were helpful and intentional. I'd be okay changing the logic flow but we need some error handling with better messaging.

Reply:
The custom exceptions that we previously had here were swallowing some of the errors I was encountering which made it hard to debug. I went ahead and added them back (see the `read_text_file` method) with some additional details to hopefully make debugging easier.